### DISCUSSION TITLE
Downloading PDF file from API response

### DISCUSSION THREAD

**Discussion by lucianrotaru** [Budibase/budibase#9438] [Help] [answered]
Hi guys! I am new here and couldn't find the answer for my problem: when clicking on a button I'm calling an API that returns a PDF file in the body of the response and I would like to download that file directly.

This is how the response looks:
![Z5ozo](https://user-images.githubusercontent.com/912615/214497435-c5770fff-c9e5-4c07-9f02-d74db5a19bfb.png)

Is there any way of downloading that file with JavaScript or something? It has to be specific a direct download without clicking on the button again.

Thank you!

**Comment by melohagan** [comment, marked as answer]
Hey @lucianrotaru 

There's an open issue here: https://github.com/Budibase/budibase/issues/8356

It may be possible to do in some "hacky" way with an Embed component, or better you could create a [custom component](https://docs.budibase.com/docs/custom-component) that allows you to download a file from an API call on click.

There's a **component** channel in the Discord server (https://discord.gg/NX6sQCYv) which could be a good place to chat with other community members and get a bit of help if you have any dev experience, or even to float the idea out there.

**Comment by tejasbirari** [comment]
Hey @lucianrotaru,

I'm also getting same error. If you have solved this error, can you please share how you resolved it ?
